### PR TITLE
Created a new page for uploading a new product

### PR DESCRIPTION
**Pull request by Unnati0509**
Created a form for adding a new product. After adding, the page is redirected to the Home page and the added product is reflected. (It still does not have cloudinary functionality, so images might not reflect on the home page.. still working on it..)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,11 +12,16 @@
   "author": "Sabeel Khan",
   "license": "ISC",
   "dependencies": {
+    "cloudinary": "^1.39.0",
```
Comment (sabeelkhan99) [inline package.json]: Please remove these dependencies from the PR if image upload is not complete

File: package.json [hunk 2]
```diff
@@ -12,11 +12,16 @@
...
     "dotenv": "^16.3.1",
     "ejs": "^3.1.9",
     "ejs-mate": "^4.0.0",
     "express": "^4.18.2",
-    "mongoose": "^7.3.4"
+    "express-fileupload": "^1.4.0",
+    "formidable": "^3.5.0",
+    "mongoose": "^7.3.4",
+    "multer": "^1.4.5-lts.1",
+    "multer-storage-cloudinary": "^4.0.0"
```
Comment (sabeelkhan99) [inline package.json]: Same applies to these dependencies as, remove it

File: src/controllers/productController.js [hunk 3]
```diff
@@ -16,9 +16,12 @@ const create = async (req, res) => {
     };
 
     const newProduct = await productService.create(product);
-    res.status(201).json(newProduct);
+    res.redirect('/api/v1/products');
 };
 
+const addProd = (req,res) => {
+    res.render('products/addProd');
+}
```
Comment (sabeelkhan99) [inline src/controllers/productController.js:24]: 1.Rename the function name such that it depicts the purpose what it does like `showNewForm`

2. Same applies to template name.

File: src/views/products/addProd.ejs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+<form action="/api/v1/products" method="POST">
+
+    <div class="mb-3">
+        <label for="imageUrl" class="form-label">Image Upload</label>
+        <input class="form-control" type="file" name="imageUrl" id="imageUrl">
+      </div>
+      <!-- <div class="mb-3">
+        <label for="image" class="form-label">Image</label>
+        <input type="text" class="form-control" name="imageUrl" id="imageUrl" placeholder="imageUrl">
+      </div> -->
```
Comment (sabeelkhan99) [inline src/views/products/addProd.ejs]: Remove these comments if not required

File: package.json (hunk 1 above)
Comment (Unnati0509) [reply package.json]: Removed the unused dependencies.

File: package.json (hunk 2 above)
Comment (Unnati0509) [reply package.json]: Removed these as well.

File: src/views/products/addProd.ejs (hunk 4 above)
Comment (Unnati0509) [reply src/views/products/addProd.ejs]: Done!

File: src/controllers/productController.js (hunk 3 above)
Comment (Unnati0509) [reply src/controllers/productController.js:24]: Done with this too, went for the same function name as stated by you.

**Pull request merged**